waflib/Tools/fc_scan.py

```diff
@@ -77,11 +90,11 @@ def iter(self, node):
 		Processes a single file during dependency parsing. Extracts files used
 		modules used and modules provided.
 		"""
+		if node in self.seen:
+			return
 		incs, uses, mods = self.find_deps(node)
+		self.seen.append(node)
```

Comment:
Not sure if this has a measurable effect but with thousands of files it might: I would consider making "seen" a set rather than a list, to make the "in" operator constant-time, rather than linear.

Reply:
I think it's important for waf to return the depends in a consistent order (ie, using a list), but this can definitely just be a set, thanks.